src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1005,6 +1042,111 @@ def _initialize(self) -> None:
                 self._backup_secrets_store.config
             )
 
+    def restore_database(self) -> None:
+        """Restore the database from a backup."""
+        # if a backup file does not exist, return
+        assert self.db_backup_file_path is not None
+        db_name = self.db_backup_file_path.split("/")[-1].split(".")[0]
+        if not os.path.exists(self.db_backup_file_path):
+            logger.debug("No backup found. No restore performed.")
+            return
+
+        with self.engine.begin() as connection:
+            # drop the database if it exists
+            connection.execute(text(f"DROP DATABASE IF EXISTS {db_name}"))
```

Comment:
Pretty sure this doesn't work on SQLite. But then again, that shouldn't need running SQL statements, it should be enough to copy the SQLite DB backup file back onto the current DB location.

Reply:
This also reminds me that in some scenarios, the database user might not be allowed to drop the database or even to create it for that matter, because it only is allowed access to this database to begin with. Not a big deal for now, just that we need to account for this in our own deployments, if we want to implement granular, per-database user access. Automated DB restoration might not work in that case.